examples/jsm/exporters/GLTFExporter.js

```diff
@@ -1067,14 +1072,7 @@ class GLTFWriter {
 
 			}
 
-			if ( ( typeof HTMLImageElement !== 'undefined' && image instanceof HTMLImageElement ) ||
-				( typeof HTMLCanvasElement !== 'undefined' && image instanceof HTMLCanvasElement ) ||
-				( typeof OffscreenCanvas !== 'undefined' && image instanceof OffscreenCanvas ) ||
-				( typeof ImageBitmap !== 'undefined' && image instanceof ImageBitmap ) ) {
-
-				ctx.drawImage( image, 0, 0, canvas.width, canvas.height );
-
-			} else {
+			if ( image.data !== undefined ) {
```

Comment:
Is this whole block meant to handle THREE.DataTexture, do you know? Had trouble reading this (especially before), let's add a comment if that's the case.

```suggestion
			if ( image.data !== undefined ) { // THREE.DataTexture
```

Reply:
Yeah, that's what this change is for. It looks like the [first commit with that code](https://github.com/mrdoob/three.js/commit/95e9206ab0d70e1f9b2fc496d2a9c03a56a4f831#diff-f90dfee333571797f4ddf7413faa1b10af7aa347ee7ec4bb10a5d4b87457f021R904) was meant to handle only DataTextures falling through to the `else` case. But in practice it has made it more brittle for other texture types like handling OffscreenCanvas textures in [this commit](https://github.com/mrdoob/three.js/commit/c029278aac6cec3ecb4f3ae7e09d4444a4c7664f).

I think the comment is a good idea for those not clear on what's going on. Thanks!